crates/goose-cli/src/prompt.rs

```diff
@@ -0,0 +1,73 @@
+use std::io::{self, Write};
+
+use anyhow::Result;
+use bat::PrettyPrinter;
+use cliclack::{input, spinner};
+use goose::providers::types::message::Message;
+
+pub trait Prompt {
+    fn render(&mut self, message: Box<Message>);
+    fn get_input(&mut self) -> Result<Input>;
+    fn show_busy(&self);
+    fn hide_busy(&self);
+    fn close(&self);
```

Comment:
Some prompt libs don't need all of these but they are important for others (eg. [ratatui](https://lib.rs/crates/ratatui))